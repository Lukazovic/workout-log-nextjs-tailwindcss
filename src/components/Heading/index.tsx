type HeadingLevels = 'h1' | 'h2' | 'h3' | 'h3' | 'h4' | 'h5' | 'h6'
type HeadingColors = 'dark' | 'light'

export type HeadingProps = {
  className?: string
  level?: HeadingLevels
  color?: HeadingColors
  children: React.ReactNode
}

const DEFAULT_STYLES = 'font-bold'

const TEXT_COLORS = {
  dark: 'text-gray-900',
  light: 'text-gray-100'
}

const Heading = ({
  className = '',
  level = 'h2',
  color = 'dark',
  children
}: HeadingProps) => {
  const fontColor = TEXT_COLORS[color]
  const style = `${`${DEFAULT_STYLES} ${className}`.trim()} ${fontColor}`.trim()

  if (level === 'h1') {
    return <h1 className={style}>{children}</h1>
  }

  if (level === 'h2') {
    return <h2 className={style}>{children}</h2>
  }

  if (level === 'h3') {
    return <h3 className={style}>{children}</h3>
  }

  if (level === 'h4') {
    return <h4 className={style}>{children}</h4>
  }

  if (level === 'h5') {
    return <h5 className={style}>{children}</h5>
  }

  return <h6 className={style}>{children}</h6>
}

export default Heading
