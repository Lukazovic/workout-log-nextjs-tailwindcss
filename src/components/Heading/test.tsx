import { render, screen } from '@testing-library/react'

import Heading from '.'

describe('<Heading />', () => {
  it('should render only a h1 heading by default', () => {
    const { container } = render(<Heading>Heading text</Heading>)

    expect(
      screen.getByRole('heading', { name: /heading text/i })
    ).toBeInTheDocument()
    expect(container.querySelectorAll('h1')).toHaveLength(1)
    expect(container.querySelectorAll('h2')).toHaveLength(0)
    expect(container.querySelectorAll('h3')).toHaveLength(0)
    expect(container.querySelectorAll('h4')).toHaveLength(0)
    expect(container.querySelectorAll('h5')).toHaveLength(0)
    expect(container.querySelectorAll('h6')).toHaveLength(0)

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render only a h2 heading', () => {
    const { container } = render(<Heading as="h2">Heading text</Heading>)

    expect(
      screen.getByRole('heading', { name: /heading text/i })
    ).toBeInTheDocument()
    expect(container.querySelectorAll('h2')).toHaveLength(1)
  })

  it('should render only a h3 heading', () => {
    const { container } = render(<Heading as="h3">Heading text</Heading>)

    expect(
      screen.getByRole('heading', { name: /heading text/i })
    ).toBeInTheDocument()
    expect(container.querySelectorAll('h3')).toHaveLength(1)
  })

  it('should render only a h4 heading', () => {
    const { container } = render(<Heading as="h4">Heading text</Heading>)

    expect(
      screen.getByRole('heading', { name: /heading text/i })
    ).toBeInTheDocument()
    expect(container.querySelectorAll('h4')).toHaveLength(1)
  })

  it('should render only a h5 heading', () => {
    const { container } = render(<Heading as="h5">Heading text</Heading>)

    expect(
      screen.getByRole('heading', { name: /heading text/i })
    ).toBeInTheDocument()
    expect(container.querySelectorAll('h5')).toHaveLength(1)
  })

  it('should render only a h6 heading', () => {
    const { container } = render(<Heading as="h6">Heading text</Heading>)

    expect(
      screen.getByRole('heading', { name: /heading text/i })
    ).toBeInTheDocument()
    expect(container.querySelectorAll('h6')).toHaveLength(1)
  })

  it('should render with dark color by default', () => {
    render(<Heading>Heading text</Heading>)

    expect(
      screen.getByRole(Heading, { name: /heading text/i })
    ).toHaveAttribute('class', 'font-bold text-4xl text-gray-900')
  })

  it('should render with light color', () => {
    render(<Heading color="light">Heading text</Heading>)

    expect(
      screen.getByRole(Heading, { name: /heading text/i })
    ).toHaveAttribute('class', 'font-bold text-4xl text-gray-100')
  })
})
