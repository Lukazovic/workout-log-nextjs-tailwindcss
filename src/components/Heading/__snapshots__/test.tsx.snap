// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render only a h1 heading by default 1`] = `
<body>
  <div>
    <h1
      class="font-bold text-4xl text-gray-900"
    >
      Heading text
    </h1>
  </div>
</body>
`;
